scripts/pylib/twister/twisterlib.py

```diff
@@ -3653,17 +3653,20 @@ def xunit_report(self, filename, platform=None, full_report=False, append=False,
                     ET.SubElement(eleTSPropetries, 'property', name="version", value=version)
 
             else:
-                eleTestsuite = ET.SubElement(eleTestsuites, 'testsuite',
-                                             name=run, time="%f" % duration,
-                                             tests="%d" % (total),
-                                             failures="%d" % fails,
-                                             errors="%d" % (errors), skipped="%s" % (skips))
-                eleTSPropetries = ET.SubElement(eleTestsuite, 'properties')
-                # Multiple 'property' can be added to 'properties'
-                # differing by name and value
-                ET.SubElement(eleTSPropetries, 'property', name="version", value=version)
+                if not no_skipped or total != skips:
```

Comment:
might be able to eliminate the double-negative if you rename the variable to `report_skipped`.